lib/attachments_client.ex

```diff
@@ -3,33 +3,40 @@ defmodule ExMicrosoftBot.Client.Attachments do
   This module provides the functions to get information related to attachments.
   """
 
-  import ExMicrosoftBot.Client, only: [authed_req_options: 1, deserialize_response: 2]
+  import ExMicrosoftBot.Client, only: [get: 1, deserialize_response: 2]
+
   alias ExMicrosoftBot.Models, as: Models
   alias ExMicrosoftBot.Client
 
   @doc """
-  Get AttachmentInfo structure describing the attachment views. [API Reference](https://docs.botframework.com/en-us/restapi/connector/#!/Attachments/Attachments_GetAttachmentInfo)
+  Get AttachmentInfo structure describing the attachment views.
+  @see [API Reference](https://docs.botframework.com/en-us/restapi/connector/#!/Attachments/Attachments_GetAttachmentInfo)
   """
-  @spec get_attachment(String.t, String.t) :: {:ok, Models.AttachmentInfo.t} | Client.error_type
+  @spec get_attachment(String.t(), String.t()) ::
+          {:ok, Models.AttachmentInfo.t()} | Client.error_type()
   def get_attachment(service_url, attachment_id) do
-    api_endpoint = "#{attachments_endpoint(service_url)}/#{attachment_id}"
-
-    HTTPotion.get(api_endpoint, authed_req_options(api_endpoint))
+    service_url
+    |> attachments_url("/#{attachment_id}")
+    |> get()
     |> deserialize_response(&Models.AttachmentInfo.parse/1)
   end
 
   @doc """
-  Get the named view as binary content. [API Reference](https://docs.botframework.com/en-us/restapi/connector/#!/Attachments/Attachments_GetAttachment)
+  Get the named view as binary content.
+  @see [API Reference](https://docs.botframework.com/en-us/restapi/connector/#!/Attachments/Attachments_GetAttachment)
   """
-  @spec get_attachment_view(String.t, String.t, String.t) :: {:ok, binary} | Client.error_type
+  @spec get_attachment_view(String.t(), String.t(), String.t()) ::
+          {:ok, String.t()} | Client.error_type()
```

Comment:
@jmnsf the return type got changed from binary to String. Was that intentional as seems like the API still returns binary.

Reply:
@zabirauf oversight, thanks!